### DISCUSSION TITLE
Realtime analysis of sweeps

### DISCUSSION THREAD

**Discussion by damianjwilliams** [AllenInstitute/MIES#2301] [Q&A] [answered]
Hi, I'm trying to plot the amplitudes of a series of sweeps in real-time. (I'm acquiring a sweep every 30 s, and would like to see how the amplitude changes over the course of the experiment). I believe that this can be done using the SweepFormula but I am getting lost in the documentation.

I can calculate the amplitude from the current trace and get the single value plotted in the SweepFormula plot, but I can't figure out how to get all the amplitudes from the previous sweeps plotted too.My code below. I'm not sure what comes next. I'd very much appreciate some guidance! Thanks

```
//define the regions of the trace where to take control and evoked measurements
t_range_control = [20,25]
t_range_evoked = [40,60]

//select the channel and all of the sweeps
sel = select(channels(AD0), sweeps())

//calculate the absolute values at control and evoked current 
dat_control  = min(data($t_range_control,$sel)
dat_evoked  = min(data($t_range_evoked,$sel)

//calculate the amplitude of the current
sweep_amplitude = $dat_evoked - $dat_control 

//plot the single amplitude value
$sweep_amplitude

```

**Comment by t-b** [comment, marked as answer]
I'm just guessing what the code is supposed to do but here is a version which plots something:

```
# define the regions of the trace where to take control and evoked measurements
t_range_control = [20,25]
t_range_evoked = [40,60]
sweepList = 1...10

# select the channel and all of the sweeps
sel = select(channels(AD0), $sweepList, all)

# calculate the absolute values at control and evoked current 
dat_control  = min(data($t_range_control,$sel))
dat_evoked  = min(data($t_range_evoked,$sel))

# calculate the amplitude of the current
sweep_amplitude = $dat_evoked - $dat_control 

# plot the single amplitude value
$sweep_amplitude
```

I changed the comment character to `#` from `//` and also tweaked the `select` statement. This now takes a list of sweeps using `all` visibility/mode. This means it does not look at the displayed sweeps which `sweeps()` returns with the implicit `displayed`.

You can get per operation help with selecting an operation name and hovering the mouse over it. When it should show you a tooltip.

For your application, plotting during data acquisition, I would enable Overlay Sweeps (OVS) and select the desired sweeps to plot. Then the `sweeps()` operation will pick up the selected sweeps and any new sweeps.

PS: We do have our SweepFormula documentation also available online at https://alleninstitute.github.io/MIES/user.html. But this is for MIES main which uses a different syntax for select and related operations.

PPS: You also might want to explore epoch information `Settings->Visualize Epochs` to not having to hardcode the extraction range but use a name like `E1` for it.

**Comment by damianjwilliams** [comment]
Hi Thomas,

Good guess as to what I was trying to accomplish! As I wanted to plot amplitudes from all the sweeps I just changed `sweepList = 1...10` to `sweepList = sweeps()` and it worked perfectly.

Many thanks,

Damian

**Comment by timjarsky** [comment]
@damianjwilliams Thank you for using MIES! May I ask how you ended up using it?

**Comment by damianjwilliams** [comment]
Hi Tim,

I ended up using it as so:

# define the regions of the trace where to take control and evoked measurements
t_range_control = [20,25]
t_range_evoked = [40,60]
sweepList = sweeps()

# select the channel and all of the sweeps
sel = select(channels(AD0), $sweepList, all)

# calculate the absolute values at control and evoked current
dat_control  = min(data($t_range_control,$sel))
dat_evoked  = min(data($t_range_evoked,$sel))

# calculate the amplitude of the current
sweep_amplitude = $dat_evoked - $dat_control

# plot the single amplitude value
$sweep_amplitude


I'm using it as a way to monitor fEPSP amplitude over the course of an
LTP experiment, in real time. I hope this answers your question!


Thanks for your help.


Damian
